br/pkg/lightning/restore/tidb_test.go

```diff
@@ -133,20 +131,11 @@ func TestCreateTableIfNotExistsStmt(t *testing.T) {
 		`, "m"))
 
 	// create view
+	// all set statements are ignored
 	require.Equal(t, []string{
-		"SET NAMES 'binary';",
 		"DROP TABLE IF EXISTS `testdb`.`m`;",
 		"DROP VIEW IF EXISTS `testdb`.`m`;",
-		"SET @`PREV_CHARACTER_SET_CLIENT`=@@`character_set_client`;",
-		"SET @`PREV_CHARACTER_SET_RESULTS`=@@`character_set_results`;",
-		"SET @`PREV_COLLATION_CONNECTION`=@@`collation_connection`;",
-		"SET @@SESSION.`character_set_client`=`utf8`;",
-		"SET @@SESSION.`character_set_results`=`utf8`;",
-		"SET @@SESSION.`collation_connection`=`utf8_general_ci`;",
```

Comment:
Is this correct for `create view` sql? I'm not so sure.

Reply:
`mydump.ExportStatment` ignores all executable comments like `/*!40014 SET FOREIGN_KEY_CHECKS=0*/`. https://github.com/pingcap/tidb/blob/4a72171ffb55b9dcf80d2dea8788ab62aff22690/br/pkg/lightning/mydump/reader.go#L108

In common cases, these set statements are written as comments. So I guess it's safe to ignore the set statements explicitly.

Reply:
These set statements are not written as comments for mydumper and [dumpling](https://github.com/pingcap/tidb/blob/ebbfb61a377ca6a787125aee55ce045f85250660/dumpling/tests/views/data/views.v-schema-view.sql#L5). They are real sqls in sql files.

Reply:
Yes. But I think they are actually equivalent to the statements in the comments. It is inconsistent to ignore one while keeping the other.

Reply:
maybe check gbk integration tests to see if `character_set_client` can be written to gbk, so lightning must execute the statement

Reply:
> maybe check gbk integration tests to see if `character_set_client` can be written to gbk, so lightning must execute the statement

The lightning_new_collation seems to pass.

Reply:
> If we allow any SQL statements in db/table schema files, it's hard for us to control the right behavior, such as charset, collate, and foreign checks. Also, it may lead to some security issues.

What about just allow `set` statement? Because it's a view sql feature. Or I think we can ignore all `set` statement in sqls but execute these `set` sqls separately for `create view` statement.

Reply:
Note that these SET statements will most likely not work with the local backend.

Reply:
> What about just allow `set` statement? Because it's a view sql feature. Or I think we can ignore all `set` statement in sqls but execute these `set` sqls separately for `create view` statement.

@lichunzhu What `set` statement is needed for view sql?

Reply:
I revert the change that ignores set statements. If the user adds some other type of SQL statement to the schema file, it is their responsibility to make sure that these statements do not affect the other logic.

@lichunzhu @lance6716 PTAL